legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/api/BaseFSApi.java

```diff
@@ -0,0 +1,174 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.api;
+
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.finos.legend.sdlc.domain.model.revision.Revision;
+import org.finos.legend.sdlc.domain.model.revision.RevisionAlias;
+import org.finos.legend.sdlc.server.api.entity.FileSystemFileAccessContext;
+import org.finos.legend.sdlc.server.api.entity.FileSystemProjectFileFileModificationContext;
+import org.finos.legend.sdlc.server.api.entity.FileSystemRevisionAccessContext;
+import org.finos.legend.sdlc.server.api.workspace.FileSystemWorkspaceApi;
+import org.finos.legend.sdlc.server.domain.api.project.source.ProjectSourceSpecification;
+import org.finos.legend.sdlc.server.domain.api.project.source.SourceSpecification;
+import org.finos.legend.sdlc.server.domain.api.project.source.SourceSpecificationVisitor;
+import org.finos.legend.sdlc.server.domain.api.project.source.WorkspaceSourceSpecification;
+import org.finos.legend.sdlc.server.exception.FSException;
+import org.finos.legend.sdlc.server.project.ProjectFileAccessProvider;
+import org.finos.legend.sdlc.server.startup.FSConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+public class BaseFSApi
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseFSApi.class);
+    protected static String rootDirectory;
+
+    public static void initRootDirectory(FSConfiguration fsConfiguration)
+    {
+        rootDirectory = fsConfiguration.getRootDirectory();
+        // Check if rootDirectory exists, and create if not
+        File localFile = new File(rootDirectory);
+        if (!localFile.exists() && !localFile.mkdirs())
+        {
+            throw new RuntimeException("Failed to create directories for rootDirectory");
+        }
+    }
+
+    public static Repository retrieveRepo(String projectId)
+    {
+        try
+        {
+            String repoDirPath = rootDirectory + File.separator + projectId + File.separator + ".git";
+            File repoDir = new File(repoDirPath);
+            if (repoDir.exists() && repoDir.isDirectory())
+            {
+                return FileRepositoryBuilder.create(repoDir);
+            }
+        }
+        catch (Exception e)
+        {
+            throw FSException.getLegendSDLCServerException("Repository " + projectId + " can't be found ", e);
+        }
+        return null;
+    }
```

Comment:
These really should not be static. Instead, consider taking FSConfiguration as a constructor argument to BaseFSApi.